types/context.go

```diff
@@ -142,7 +152,23 @@ func (c Context) MustGetPrevCtx(height int64) Context {
 	return con
 }
 
+func (c Context) getFromCache(key string) (interface{}, bool) {
+	if c.CachedStore == nil {
+		return nil, false
+	}
+	return c.CachedStore.Get(key)
+}
+func (c Context) addToCache(key string, i interface{}) (evicted bool) {
+	if c.CachedStore == nil {
+		return false
+	}
+	return c.CachedStore.Add(key, i)
+}
+
 func (c Context) PrevCtx(height int64) (Context, error) {
+	if cachedCtx, ok := c.getFromCache(fmt.Sprintf("PrevCtx%v", height)); ok {
```

Comment:
Since we are trying to shave every bit off of memory usage, let's remove `PrevCtx` from the key string and just use the height... Could even use bytes if that's better

Reply:
simple-ru only supports string keys we could just use the height as key